File: 38.count-and-say.py
#
# @lc app=leetcode id=38 lang=python3
#
# [38] Count and Say
#

# @lc code=start
class Solution:
    def countAndSay(self, n: int) -> str:
        
        return self.say(n, "1")

    def say(self, n: int, string: str):
        if n == 1:
            return string

        count = 1
        current = string[0]
        result = ""
        for each in string[1:]:
            if current == each:
                count += 1
            else:
                result += (str(count) + str(current))
                count = 1
                current = each
        result += (str(count) + str(current))

        return self.say(n - 1, result)

# @lc code=end

s = Solution()
print(s.countAndSay(4))

File: 38.count-and-say.js
/*
 * @lc app=leetcode id=38 lang=javascript
 *
 * [38] Count and Say
 */

// @lc code=start
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {

  var string = '1';

  for (var i = 2; i <= n; i++) {
    var temp = '';

    var char = string[0];
    var count = 1;
    for (var j = 1; j < string.length; j++) {
      if (string[j] != char) {
        temp = temp + count.toString() + char;
        count = 1;
        char = string[j];
      } else {
        count++;
      }
    }
    string = temp + count.toString() + char;
  }

  return string;
};
// @lc code=end

